src/PluginsManager/Microsoft.Performance.Toolkit.Plugins.Runtime/Common/IRepositoryRO`1.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.Specialized;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime.Common
+{
+    /// <summary>
+    ///     Represents a repository for storing a collection of entities of type <typeparamref name="TEntity"/>.
+    ///     Supports read-only operations and notifications when the repository is modified.
+    /// </summary>
+    /// <typeparam name="TEntity"></typeparam>
+    public interface IRepositoryRO<TEntity>
```

Comment:
I'm realizing this interface is essentially the same thing as `IReadOnlyObservableCollection`. Do we want to just use that, or keep this?

Reply:
Looks like  .NET doesn't have the class but no interface for `ObservableCollection` .  WPA do have them under the `Microsoft.VisualStudio.PlatformUI.Shell`